### PR TITLE
Increase and clean warnings 

### PR DESCRIPTION
**Pull request by galanCA**
Increases the warning level and cleans all the new warnings.

### CODE REVIEW

File: src/gl/context.cpp [hunk 1]
```diff
@@ -127,7 +127,7 @@ Context::Context(unsigned size)
   };
 
   auto scroll_callback = [](GLFWwindow *w, double /*xOffset*/, double yOffset) {
-    glfwWPtr(w)->viewScale += 0.1f * yOffset;
+    glfwWPtr(w)->viewScale += 0.1f * static_cast<float>(yOffset);
```
Comment (chipbarnaby) [inline src/gl/context.cpp]: Prefer static_cast<float>( 0.1 * yOffset)

Do the calc as double, then cast the result.

(Minor impact if any, but still ... . )

**Review by chipbarnaby — APPROVED** [review]
Looks fine.  I made one minor comment.

File: src/gl/context.cpp (hunk 1 above)
Comment (galanCA) [reply src/gl/context.cpp]: Out of curiosity, is doing the calculation and then casting faster than the opposite way?

**Comment by galanCA** [comment]
Thank you @chipbarnaby for the review.

**Review by nealkruis — APPROVED** [review]

**Pull request merged**